packages/brokeneck-react/src/components/Navigation.js

```diff
@@ -6,49 +6,95 @@ import {
   IconButton,
   makeStyles,
   Tab,
-  Tabs,
-  Typography
+  Tabs
 } from '@material-ui/core'
+import { useTheme } from '@material-ui/core/styles'
+import GroupIcon from '@material-ui/icons/Group'
+import UserIcon from '@material-ui/icons/Person'
+
+import MoonIcon from '../icons/moon'
+import SunIcon from '../icons/sun'
 
 import { ThemeSwitcherContext } from './ThemeSwitcherProvider'
 import Provider from './Provider'
 
 const useStyles = makeStyles(theme => ({
   appBar: {
+    alignItems: 'center',
+    backgroundColor: theme.palette.headerBackground.main,
+    color: theme.palette.bodyText.main,
+    flexDirection: 'row',
     marginBottom: theme.spacing(3),
-    flexDirection: 'row'
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(4)
+  },
+  provider: {
+    marginLeft: 'auto'
   },
-  actions: {
-    marginLeft: 'auto',
-    '& > * + *': {
-      marginLeft: theme.spacing(1)
-    }
+  switcher: {
+    borderLeft: `1px solid ${theme.palette.separator.main}`,
+    borderRight: `1px solid transparent`, // Needed for symetry
+    height: '42px',
+    marginLeft: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    width: '42px'
+  },
+  tabRoot: {
+    minWidth: 120
+  },
+  tabWrapper: {
+    flexDirection: 'row'
   }
 }))
 
 export default function Navigation() {
   const classes = useStyles()
+  const {
+    palette: { type: themeType }
+  } = useTheme()
   const toggleThemeType = useContext(ThemeSwitcherContext)
```

Comment:
let's encapsulate this in ThemeSwitcherContext and return an object from it instead of mixing useTheme and useContext. all theme switching stuff should come from ThemeSwitcherContext, meaning both the current theme type and the toggleThemeType function